packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -0,0 +1,443 @@
+      <Carousel visibleItems={2} onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+
+    const nextButton = getByRole('button', { name: 'Next Slide' });
+    const previousButton = getByRole('button', { name: 'Previous Slide' });
+    fireEvent.click(nextButton);
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+
+    fireEvent.click(previousButton);
+
+    expect(onChange).toHaveBeenLastCalledWith(0);
+
+    // assert indicator button count
+    expect(queryAllByRole('tab').length).toBe(3);
+  });
+
+  it('should auto play', async () => {
+    jest.useFakeTimers();
+    const onChange = jest.fn();
+    const { queryAllByTestId } = renderWithTheme(
+      <Carousel autoPlay visibleItems={1} onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(2);
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(0);
+  });
+
+  it('should not auto play when mouse is over', async () => {
+    jest.useFakeTimers();
+    const onChange = jest.fn();
+    const { getByLabelText, queryAllByTestId } = renderWithTheme(
+      <Carousel accessibilityLabel="My Carousel" autoPlay visibleItems={1} onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+
+    // hover over carousel
+    fireEvent.mouseEnter(getByLabelText('My Carousel'));
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
```

Comment:
shouldn't this rather be `toHaveNotBeenCalled()`?

Reply:
Yeah ideally yes, we said to only call this on user interaction. 

I tried fixing this, it is getting a bit tricky to know when the carousel is scrolling via user's touch & drag vs when it's scrolling via autoScroll. 

Should we just keep it as is and call it on state change, be it user interaction or autoPlay?

Reply:
state change as in?

Reply:
just as is, jaise hain abhi. 

React.useEffect(() => { onChange(activeIndex) }, [activeIndex])